### DISCUSSION TITLE
Recommended nodejs version?

### DISCUSSION THREAD

**Discussion by sodiumchl** [ONLYOFFICE/build_tools#432] [Q&A] [answered]
What is the recommend nodejs version for OO v7? 

- `automate.py` installs v10
- `apt install documentserver` pulls in v12 ubuntu 20.04
- The latest  LTS is v16.

**Comment by ShockwaveNN** [comment]
We use v10 in `automate.py` to be compatible with some rare OS'es we need to work, but I think we in process of actualizing it

Generally speaking - you can use any version you want, I'm not aware that there are any problems related to `nodejs` version at all


> apt install documentserver pulls in v12 ubuntu 20.04

This is not right, DocumentServer do not require any `nodejs` any more, since we use [`pkg`](https://github.com/vercel/pkg) to package our services and don't require `nodejs` at all

**Reply by sodiumchl** [reply]
It seems node14 is set in `build_tools/scripts/build_server.py`:
`pkg_target = "node14"`
`base.cmd_in_dir(server_build_dir + "/FileConverter", "pkg", [".", "-t", pkg_target, "-o", "converter"])`

Does this mean onlyoffice/documentserver should be built with node14?

> DocumentServer do not require any nodejs any more
You are right. It was from somewhere else.

**Reply by ShockwaveNN** [reply, marked as answer]
As I said - you can use any version of nodejs newer that 10.22 (If I remember right) and everything should work fine

```
pkg_target = "node14"
base.cmd_in_dir(server_build_dir + "/FileConverter", "pkg", [".", "-t", pkg_target, "-o", "converter"])
```

I'm not very good at `pkg` but as we tell `pkg` to use it's internal nodejs versioning mechanism to build resulting executable with noidejs 14 framework

**Reply by sodiumchl** [reply]
I see. Thank you!
